skbuild/resources/cmake/targetLinkLibrariesWithDynamicLookup.cmake

```diff
@@ -198,6 +198,28 @@ function(_test_weak_link_project
   set(osx_dynamic_lookup           "-undefined dynamic_lookup")
   set(no_flag                                               "")
 
+  if(CMAKE_CROSSCOMPILING)
+    set(link_flag_spec "no_flag")
+    set(link_flag "${link_flag_spec}")
+    set(test_skipping_reason "")
+    set(test_pass FALSE)
+
+    if(APPLE AND NOT CMAKE_CROSSCOMPILING_EMULATOR)
+      set(link_flag_spec "osx_dynamic_lookup")
+      set(link_flag "${link_flag_spec}")
```

Comment:
So this doesn't work, the cross-build errors with `error: no such file or directory: 'osx_dynamic_lookup'`.  I think that there's something here with a string replacement that isn't happening (it should be substituting `-undefined dynamic_lookup`) but something with quotes/brackets/?? is going wrong.

Reply:
Thanks for testing, the next commit should address this.

_Note that once we have confirmed it works, commits will be squashed_